_posts/2023-03-29-langchain-slackbot.md

```diff
@@ -0,0 +1,104 @@
+
+**Evolving Best Practices**: In the rapidly evolving world of LLMOps, there are no hard and fast rules yet. Best practices are still being defined as the field advances and professionals gain more experience with large language models. This necessitates flexibility and adaptability when working on such projects.
+
+**Rapid Development of Frameworks and Tooling**: The pace of development in the field is extremely fast, with frameworks and tools being built and improved continuously. Community contributions to libraries like `langchain` and `llama_index` have been instrumental in driving progress. However, it's essential to recognize that the LLM world still has a long way to go before reaching 'mature' tooling.
+
+**Data Remains King**: As with most machine learning applications, data plays a crucial role in the success of LLMs. For instance, importing Slack support conversations into the vector store led to less robust and useful responses from the Slack bot. Ensuring the quality and relevance of the data used for training and fine-tuning LLMs is vital.
+
+**Efficient Querying**: There is significant scope for developing smart and efficient ways to query data in LLM applications. It is important to minimize the amount of data sent to external endpoints, such as OpenAI's, to reduce costs associated with token usage and to maintain data privacy by avoiding exposing real user data on third-party infrastructure.
+
+**Adapting Deployment Best Practices**: While there are existing best practices for deploying text-based tooling, the growth of LLM usage will likely prompt the need to learn and adapt these practices to accommodate new experiments and innovations. As more professionals explore the potential of LLMs, the collective knowledge and experience will contribute to refining best practices and guiding future developments in the field.
+
+We benefited from many of ZenML's core features during this project:
+
+- caching allowed us to run our pipelines without needing to wait excessively to repeatedly query the same (mostly-unchanging) documents or data sources
+- switching between local stacks and deployed cloud infrastructure was simple, as always!
+- using ZenML's post-execution workflow we were able to query the artifact store and get the de-serialised FAISS vector store. This we then used within our deployed slack bot app.
+- collaboration through the dashboard: colleagues were able to inspect the progress of our pipelines as well as see where the underlying data was stored through the ZenML dashboard.
+
+## Where to go from here?
+
+If you'd like to see our Slack bot in action, feel free to [join us in Slack](https://zenml.io/slack) and get help on your problems even when the core team are asleep! Just mention `@zenml-bot` and you'll be on your way! The code behind the project is all available [in our open-source Projects repository](https://zenml.io/projects), as usual, and we'd invite you to check out how this all works under the hood.
```

Comment:
Might want to link to the github here actually not the website? You can say all similar projects are here: https://zenml.io/projects